[GraalVM] [Nats] Problems when running the application

### Task List

- [ ] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [ ] Example that reproduces the problem uploaded to Github
- [ ] Full description of the issue provided (see below)

### Steps to Reproduce

1. The application is built sucessfully.
```
❯ ./gradlew nativeImage

> Task :compileJava
Note: Writing resource-config.json file to destination: META-INF/native-image/io.x/core-cash-in-adapter/resource-config.json
Note: Creating bean classes for 3 type elements

> Task :nativeImage
[application:23362]    classlist:   1,601.51 ms,  0.96 GB
[application:23362]        (cap):   1,670.97 ms,  0.96 GB
[application:23362]        setup:   2,704.44 ms,  0.96 GB
[application:23362]     (clinit):   1,126.40 ms,  4.73 GB
[application:23362]   (typeflow):  17,077.90 ms,  4.73 GB
[application:23362]    (objects):  16,269.12 ms,  4.73 GB
[application:23362]   (features):   2,299.96 ms,  4.73 GB
[application:23362]     analysis:  38,694.73 ms,  4.73 GB
[application:23362]     universe:   4,412.22 ms,  5.29 GB
[application:23362]      (parse):   4,827.51 ms,  5.29 GB
[application:23362]     (inline):  13,728.68 ms,  6.36 GB
[application:23362]    (compile):  28,372.86 ms,  6.06 GB
[application:23362]      compile:  49,979.60 ms,  6.06 GB
[application:23362]        image:   5,537.61 ms,  5.86 GB
[application:23362]        write:   1,602.33 ms,  5.86 GB
[application:23362]      [total]: 104,688.23 ms,  5.86 GB
Native Image written to: <REDACTED>/build/native-image/application
```
2. And then run the application:
```
11:26:10.015 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Error processing bean [Definition: io.x.handler.NatsCashInListener] method definition [CashInResponse receiveCashInMessage(CashInRequest cashInRequest)]: Error instantiating bean of type [io.nats.client.Connection]: java.lang.ClassNotFoundException: io.nats.client.impl.SocketDataPort
io.micronaut.context.exceptions.BeanContextException: Error processing bean [Definition: io.x.handler.NatsCashInListener] method definition [CashInResponse receiveCashInMessage(CashInRequest cashInRequest)]: Error instantiating bean of type [io.nats.client.Connection]: java.lang.ClassNotFoundException: io.nats.client.impl.SocketDataPort
        at io.micronaut.context.AnnotationProcessorListener.onCreated(AnnotationProcessorListener.java:92)
        at io.micronaut.context.AnnotationProcessorListener.onCreated(AnnotationProcessorListener.java:45)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1955)
        at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:3022)
        at io.micronaut.context.DefaultBeanContext.getBeansOfTypeInternal(DefaultBeanContext.java:2897)
        at io.micronaut.context.DefaultBeanContext.streamOfType(DefaultBeanContext.java:772)
        at io.micronaut.context.DefaultBeanContext.streamOfType(DefaultBeanContext.java:759)
        at io.micronaut.context.DefaultBeanContext.lambda$initializeContext$32(DefaultBeanContext.java:1596)
        at java.util.HashMap.forEach(HashMap.java:1336)
        at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1595)
        at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:220)
        at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2812)
        at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:233)
        at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:166)
        at io.micronaut.runtime.Micronaut.start(Micronaut.java:64)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:299)
        at io.micronaut.runtime.Micronaut.run(Micronaut.java:285)
        at io.x.Application.main(Application.java:8)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type [io.nats.client.Connection]: java.lang.ClassNotFoundException: io.nats.client.impl.SocketDataPort
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1927)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2679)
        at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2665)
        at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2337)
        at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2311)
        at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:721)
        at io.micronaut.nats.intercept.NatsConsumerAdvice.process(NatsConsumerAdvice.java:105)
        at io.micronaut.nats.intercept.NatsConsumerAdvice.process(NatsConsumerAdvice.java:54)
        at io.micronaut.context.AnnotationProcessorListener.onCreated(AnnotationProcessorListener.java:89)
        ... 17 common frames omitted
Caused by: java.lang.IllegalArgumentException: java.lang.ClassNotFoundException: io.nats.client.impl.SocketDataPort
        at io.nats.client.Options$Builder.createInstanceOf(Options.java:719)
        at io.nats.client.Options.buildDataPort(Options.java:1323)
        at io.nats.client.impl.NatsConnection.tryToConnect(NatsConnection.java:427)
        at io.nats.client.impl.NatsConnection.connect(NatsConnection.java:217)
        at io.nats.client.impl.NatsImpl.createConnection(NatsImpl.java:29)
        at io.nats.client.Nats.createConnection(Nats.java:271)
        at io.nats.client.Nats.connect(Nats.java:186)
        at io.micronaut.nats.connect.NatsConnectionFactory.connection(NatsConnectionFactory.java:53)
        at io.micronaut.nats.connect.$NatsConnectionFactory$Connection0Definition.build(Unknown Source)
        at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:148)
        at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1898)
        ... 25 common frames omitted
Caused by: java.lang.ClassNotFoundException: io.nats.client.impl.SocketDataPort
        at com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:60)
        at java.lang.Class.forName(DynamicHub.java:1279)
        at io.nats.client.Options$Builder.createInstanceOf(Options.java:715)
        ... 35 common frames omitted

```
3. TODO

### Expected Behaviour


### Actual Behaviour


### Environment Information

- **Operating System**: Linux
- **Micronaut Version:** 2.2.2
- **JDK Version:** 11

### Example Application

--- a/nats/src/main/resources/META-INF/native-image/io/micronaut/nats/native-image.properties
+++ b/nats/src/main/resources/META-INF/native-image/io/micronaut/nats/native-image.properties
@@ -1,0 +1,2 @@
+Args = --initialize-at-build-time=io.nats.client.impl.NatsConnection \
+       --initialize-at-run-time=io.micronaut.nats.health.$NatsHealthIndicatorDefinition

--- a/nats/src/main/resources/META-INF/native-image/io/micronaut/nats/reflect-config.json
+++ b/nats/src/main/resources/META-INF/native-image/io/micronaut/nats/reflect-config.json
@@ -1,0 +1,6 @@
+[
+{
+  "name":"io.nats.client.impl.SocketDataPort",
+  "methods":[{"name":"<init>","parameterTypes":[] }]
+}
+]

